### PR TITLE
TF-1606 Handle relative url in session

### PR DESCRIPTION
**Pull request by dab246**
### Issue

#1606

### CODE REVIEW

**Review by nqhhdev — APPROVED** [review]
Lgtm

**Review by chibenwa — APPROVED** [review]
Works.

THANKS

File: core/lib/presentation/extensions/uri_extension.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    log('SessionUtils::toQualifiedUrl():sourceUrl: $this');
+    if (toString().startsWith(baseUrl.toString())) {
+      final qualifiedUrl = toString().removeLastSlashOfUrl();
+      log('SessionUtils::toQualifiedUrl():qualifiedUrl: $qualifiedUrl');
+      return Uri.parse(qualifiedUrl);
+    } else {
+      final baseUrlValid = baseUrl.toString().removeLastSlashOfUrl();
+      final sourceUrlValid = toString().addFirstSlashOfUrl().removeLastSlashOfUrl();
+      log('SessionUtils::toQualifiedUrl():baseUrlValid: $baseUrlValid');
+      log('SessionUtils::toQualifiedUrl():sourceUrlValid: $sourceUrlValid');
```
Comment (hoangdat) [inline core/lib/presentation/extensions/uri_extension.dart:17]: idem

File: core/test/presentation/extensions/uri_extension_test.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
+
+import 'package:core/presentation/extensions/uri_extension.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+
+  group('method convertToQualifiedUrl test', () {
+
+    test('convertToQualifiedUrl() should return qualified url when baseUrl is `https://domain.com` and sourceUrl is `https://domain.com/jmap`', () async {
+      final baseUrl =  Uri.parse('https://domain.com');
```
Comment (hoangdat) [inline core/test/presentation/extensions/uri_extension_test.dart:10]: You mean: `baseUrl` = what user input? Am I right?

Comment (dab246) [reply core/test/presentation/extensions/uri_extension_test.dart:10]: It is jmapUrl

**Pull request merged**